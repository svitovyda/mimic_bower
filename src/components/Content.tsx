import { useContentFetcher } from "../hooks/contentFetcher";
import { ModuleItem } from "./ModuleItem";
import { LoaderAnimation } from "./ui/LoaderAnimation";
import * as React from "react";

export const Content: React.FC = () => {
  const [state, setQuery] = useContentFetcher("");

  React.useEffect(() => {
    setQuery("jquery");
  }, [setQuery]);

  if (state.isLoading) return <LoaderAnimation />;
  if (state.isError) return <div>Some error occurred, try again later!</div>;
  return (
    <div>
      {state.data.map((p) => (
        <ModuleItem key={p.name} packageElement={p} />
      ))}
    </div>
  );
};
