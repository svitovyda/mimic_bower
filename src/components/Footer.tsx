import * as React from "react";

export const Footer: React.FC = () => {
  return (
    <footer>
      <p>
        <a href="https://github.com/bower/bower.github.io/issues" target="_blank" rel="noreferrer">
          Help improve these docs. Open an issue or pull request.
        </a>
      </p>
      <p>Our supporters: TBA</p>
    </footer>
  );
};
