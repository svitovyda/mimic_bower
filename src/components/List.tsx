import type { Package } from '../models/Package';
import { ModuleItem } from './ModuleItem';
import * as React from 'react';

export interface ListProps {
  data: Package[];
}

export const List: React.FC<ListProps> = ({ data }: ListProps) => {
  return (
    <div>
      {data.map((p, i) => (
        <ModuleItem key={`${i}-${p.name}-${p.platform}`} packageElement={p} />
      ))}
    </div>
  );
};
