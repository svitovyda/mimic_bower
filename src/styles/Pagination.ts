import styled from 'styled-components';

export const PaginationContainer = styled.div`
  margin-top: 10px;
`;
