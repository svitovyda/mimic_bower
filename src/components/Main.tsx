import { Content } from "./Content";
import { Search } from "./Search";
import * as React from "react";

export const Main: React.FC = () => {
  const [query, setQuery] = React.useState<string>("");

  const onNewQuery = React.useCallback(
    (newQuery: string) => {
      setQuery(newQuery);
    },
    [setQuery]
  );

  return (
    <div>
      <Search onQueryChanged={onNewQuery} />
      <Content query={query} />
    </div>
  );
};
