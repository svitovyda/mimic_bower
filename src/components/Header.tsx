import { HeaderContainer, HeaderLogo, HeaderSubtitle, HeaderTitle } from '../styles/Header';
import * as React from 'react';

export const Header: React.FC = () => {
  return (
    <header>
      <HeaderContainer>
        <HeaderLogo />
        <HeaderTitle>Bower Search</HeaderTitle>
        <HeaderSubtitle>
          Powered by{' '}
          <a href="https://libraries.io/" target="_blank" rel="noreferrer">
            libraries.io
          </a>
        </HeaderSubtitle>
      </HeaderContainer>
    </header>
  );
};
