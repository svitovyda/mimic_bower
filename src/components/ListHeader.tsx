/* eslint-disable jsx-a11y/anchor-is-valid */
import type { SortType } from '../services/ApiServise';
import { ListHeaderContainer, NameLink, OwnerLink, StarsLink } from '../styles/ListHeader';
import * as React from 'react';

export interface ListHeaderProps {
  onSort: (sortBy?: SortType) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = React.memo(({ onSort }: ListHeaderProps) => {
  return (
    <ListHeaderContainer>
      <NameLink onClick={() => onSort()}>Name</NameLink>
      <OwnerLink>Owner</OwnerLink>
      <StarsLink onClick={() => onSort('stars')}>Stars</StarsLink>
    </ListHeaderContainer>
  );
});

ListHeader.displayName = 'ListHeader';
