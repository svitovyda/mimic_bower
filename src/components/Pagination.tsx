import { PaginationButton } from "../styles/Pagination";
import * as React from "react";

export interface PaginationProps {
  totalPages: number;
  currentPage: number;
  navigateToPage: (page: number) => void;
}
export const Pagination: React.FC<PaginationProps> = ({ totalPages, currentPage, navigateToPage }: PaginationProps) => {
  const handleNext = () => {
    if (currentPage < totalPages) {
      navigateToPage(currentPage + 1);
    }
  };

  const handlePrev = () => {
    if (currentPage > 1) {
      navigateToPage(currentPage - 1);
    }
  };

  return (
    <div>
      <PaginationButton disabled={currentPage === 1} onClick={handlePrev}>
        ← Previous
      </PaginationButton>
      <PaginationButton disabled={currentPage === totalPages || totalPages === 0} onClick={handleNext}>
        Next →
      </PaginationButton>
    </div>
  );
};
