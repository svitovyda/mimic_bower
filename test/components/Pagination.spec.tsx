import { Pagination } from "../../src/components/Pagination";
import { render } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";

describe("Pagination", () => {
  it("renders without error for initial data", () => {
    const component = render(<Pagination currentPage={1} totalPages={0} navigateToPage={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });

  it("renders without error for invalid data", () => {
    const component = render(<Pagination currentPage={-1} totalPages={-10} navigateToPage={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });

  it("renders without error for valid data", () => {
    const component = render(<Pagination currentPage={5} totalPages={10} navigateToPage={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });

  it("renders last page", () => {
    const component = render(<Pagination currentPage={10} totalPages={10} navigateToPage={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });
});
