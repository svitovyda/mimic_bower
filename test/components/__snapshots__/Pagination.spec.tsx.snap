// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders last page 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<body>
    <div>
      <div
        class="c0"
      >
        <button
          class="c1"
        >
          ← Previous
        </button>
        <button
          class="c1"
          disabled=""
        >
          Next →
        </button>
      </div>
    </div>
  </body>,
  "container": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        ← Previous
      </button>
      <button
        class="c1"
        disabled=""
      >
        Next →
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pagination renders without error for initial data 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<body>
    <div>
      <div
        class="c0"
      >
        <button
          class="c1"
          disabled=""
        >
          ← Previous
        </button>
        <button
          class="c1"
          disabled=""
        >
          Next →
        </button>
      </div>
    </div>
  </body>,
  "container": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<div>
    <div
      class="c0"
    >
      <button
        class="c1"
        disabled=""
      >
        ← Previous
      </button>
      <button
        class="c1"
        disabled=""
      >
        Next →
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pagination renders without error for invalid data 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<body>
    <div>
      <div
        class="c0"
      >
        <button
          class="c1"
        >
          ← Previous
        </button>
        <button
          class="c1"
        >
          Next →
        </button>
      </div>
    </div>
  </body>,
  "container": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        ← Previous
      </button>
      <button
        class="c1"
      >
        Next →
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pagination renders without error for valid data 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<body>
    <div>
      <div
        class="c0"
      >
        <button
          class="c1"
        >
          ← Previous
        </button>
        <button
          class="c1"
        >
          Next →
        </button>
      </div>
    </div>
  </body>,
  "container": .c0 {
  margin-top: 10px;
}

.c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

<div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        ← Previous
      </button>
      <button
        class="c1"
      >
        Next →
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
